UMassCDS/inatator
# Update components with Mantine Component Library

## Added
- mantine component library components
- frontend is switched to vite+react, recommended by mantine library
- Application now uses Vite build tool
- Added new visuals for loading status
## Updated
- README updated to reflect new application url and instruction on running Vite
- Converted old JS components into JSX components and updated components with mantine counterparts
- frontend dockerfile to build vite before serving
- Broke up functions in App.js into separate functions in utils.js, makes it more readable
## Removed
- old css files that are now unused
- loadingbarstatus.js, now we use loading overlays and display errors with alerts

## How to test
You can still follow regular instructions in readme to start local instance, docker is not working because of backend-postgres connection, explained below.

Two important things: make sure to run `npm i` and then `npm run dev` to start the application, it will be served in `localhost:5173`

## Problems
I noticed that docker-compose removed env-file option, I was not able to run backend because it needs a specific url, I think this was done for deployment, or I just have incorrect url in my project configurations

Leaflet has a known issue: https://github.com/Leaflet/Leaflet.draw/issues/1026, it is not patched yet and causes us to create a global variable to not the throw an error

Closes #85
